jinahub/indexers/simple/__init__.py

```diff
@@ -0,0 +1,139 @@
+        self._docs = DocumentArrayMemmap(self.workspace + f'/{index_file_name}')
+        self.default_traversal_paths = default_traversal_paths or ['r']
+        self.default_top_k = default_top_k
+        if distance_metric == 'cosine':
+            self.distance = _cosine
+        elif distance_metric == 'euclidean':
+            self.distance = _euclidean
+        else:
+            raise ValueError('This distance metric is not available!')
+        self._flush = True
+        self._docs_embeddings = None
+
+    @property
+    def index_embeddings(self):
+        if self._flush:
+            self._docs_embeddings = np.stack(self._docs.get_attributes('embedding'))
+            self._flush = False
+        return self._docs_embeddings
+
+    @requests(on='/index')
+    def index(self, docs: 'DocumentArray', parameters: Dict, **kwargs):
+        """All Documents to the DocumentArray
+        :param docs: the docs to add
+        :param parameters: the parameters dictionary
+        """
+        traversal_path = parameters.get('traversal_paths', self.default_traversal_paths)
+        flat_docs = docs.traverse_flat(traversal_path)
+        self._docs.extend(flat_docs)
+        self._flush = True
+
+    @requests(on='/search')
+    def search(self, docs: 'DocumentArray', parameters: Dict, **kwargs):
+        """Perform a vector similarity search and retrieve the full Document match
+
+        :param docs: the Documents to search with
+        :param parameters: the parameters for the search"""
+        traversal_path = parameters.get('traversal_paths', self.default_traversal_paths)
+        top_k = parameters.get('top_k', self.default_top_k)
+        flat_docs = docs.traverse_flat(traversal_path)
+        a = np.stack(flat_docs.get_attributes('embedding'))
+        b = self.index_embeddings
+        q_emb = _ext_A(_norm(a))
+        d_emb = _ext_B(_norm(b))
+        dists = self.distance(q_emb, d_emb)
+        idx, dist = self._get_sorted_top_k(dists, int(top_k))
+        for _q, _ids, _dists in zip(flat_docs, idx, dist):
+            for _id, _dist in zip(_ids, _dists):
+                d = Document(self._docs[int(_id)], copy=True)
+                d.scores['cosine'] = 1 - _dist
+                _q.matches.append(d)
+
+    @staticmethod
+    def _get_sorted_top_k(
+        dist: 'np.array', top_k: int
+    ) -> Tuple['np.ndarray', 'np.ndarray']:
+        if top_k >= dist.shape[1]:
+            idx = dist.argsort(axis=1)[:, :top_k]
+            dist = np.take_along_axis(dist, idx, axis=1)
+        else:
+            idx_ps = dist.argpartition(kth=top_k, axis=1)[:, :top_k]
+            dist = np.take_along_axis(dist, idx_ps, axis=1)
+            idx_fs = dist.argsort(axis=1)
+            idx = np.take_along_axis(idx_ps, idx_fs, axis=1)
+            dist = np.take_along_axis(dist, idx_fs, axis=1)
+
+        return idx, dist
+
+    @requests(on='/fill_embedding')
+    def fill_embedding(self, query_da: DocumentArray, **kwargs):
```

Comment:
Only `docs` are available. https://github.com/jina-ai/jina/blob/master/.github/2.0/cookbooks/Executor.md#method-arguments
```suggestion
    def fill_embedding(self, docs: DocumentArray, **kwargs):
```

Reply:
Done